### PR TITLE
Issue #3309: Remove login tabs

### LINKED ISSUE #3309: [WP][UX] Remove "tabs" from log-in, register, and password reset pages.
A follow-up to https://github.com/backdrop/backdrop-issues/issues/3108

I'd really like to redesign the log-in page to remove the "tabs" entirely. The visual "tabs" are a strange way to navigate these three pages. Plus, they really give away that a site is Drupal/Backdrop.

I almost always remove these tabs in my custom themes, and replace them with a markup element in the form containing links to those pages instead. I'd love to see us do that by default for all Backdrop sites.

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/3309

### CODE REVIEW

File: core/modules/user/user.admin.inc [hunk 1]
```diff
@@ -399,14 +399,41 @@ function user_login_settings($form, &$form_state) {
...
   );
 
+  $form['user_login_page_settings'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Login page settings'),
+  );
+  $form['user_login_page_settings']['user_login_page_tabs'] = array(
+    '#title' => t('Use tabs'),
+    '#type' => 'checkbox',
+    '#description' => t('If selected shows local tabs for "Create new account", "Log in", and "Reset passord" links. A simplified page with links is shown if unchecked.'),
```
Comment (klonos) [inline core/modules/user/user.admin.inc:409]: Typo:
```suggestion
    '#description' => t('If selected shows local tabs for "Create new account", "Log in", and "Reset password" links. A simplified page with links is shown if unchecked.'),
```

File: core/modules/user/user.admin.inc [hunk 2]
```diff
@@ -399,14 +399,41 @@ function user_login_settings($form, &$form_state) {
     ),
   );
 
+  $form['user_login_page_settings'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Login page settings'),
```
Comment (klonos) [inline core/modules/user/user.admin.inc:404]:
```suggestion
    '#title' => t('Login page settings'),
    '#description' => t('These settings will affect the style of the "Log in", "Create new account", and "Reset password" pages.'),
```

File: core/modules/user/user.admin.inc (hunk 1 above)
Comment (klonos) [reply core/modules/user/user.admin.inc:409]: ...however, if this was converted to a `Login links style` radio set with two options, this lengthy help text would either not be required, or be repurposed to explain details/differences about the two options. Something like this:
- Horizontal tabs
  > Typically placed over the login form.
- Links
  > Typically placed below the login form.

File: core/modules/user/user.admin.inc [hunk 3]
```diff
@@ -399,14 +399,41 @@ function user_login_settings($form, &$form_state) {
...
+  );
+  $form['user_login_page_settings']['user_login_page_tabs'] = array(
+    '#title' => t('Use tabs'),
+    '#type' => 'checkbox',
+    '#description' => t('If selected shows local tabs for "Create new account", "Log in", and "Reset passord" links. A simplified page with links is shown if unchecked.'),
+    '#default_value' => $core_config->get('user_login_page_tabs'),
+    '#config' => 'system.core',
```
Comment (klonos) [inline core/modules/user/user.admin.inc:411]: Perhaps move this to the container fieldset?

File: core/modules/user/user.module [hunk 4]
```diff
@@ -1411,26 +1411,27 @@ function user_menu() {
...
   );
 
+  $user_login_page_tabs = user_login_page_tabs();
   $items['user/login'] = array(
     'title' => 'Log in',
     'page callback' => 'backdrop_get_form',
     'page arguments' => array('user_login'),
     'access callback' => 'user_is_anonymous',
-    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'type' => ($user_login_page_tabs ? MENU_DEFAULT_LOCAL_TASK : MENU_CALLBACK),
```
Comment (indigoxela) [inline core/modules/user/user.module:1420]: The position of the closing parenthesis is irritating.
Also two times below, when it's used the same way.

File: core/modules/user/user.module [hunk 5]
```diff
@@ -1630,6 +1631,19 @@ function user_menu_alter(&$items) {
   }
 }
 
+/**
+ * Return the user page tabs setting.
+ */
+function user_login_page_tabs() {
+  static $user_login_page_tabs;
```
Comment (indigoxela) [inline core/modules/user/user.module:1638]: Why a static cache? Is this supposed to be called many times per page request?

File: core/profiles/standard/standard.install [hunk 6]
```diff
@@ -586,6 +586,7 @@ function standard_install() {
     'link_path' => 'user/login',
     'menu_name' => 'user-menu',
     'weight' => -1,
+    'customized' => 1,
```
Comment (indigoxela) [inline core/profiles/standard/standard.install:589]: What's actually the purpose of this "customized" here? Can't seem to find any reference anywhere else in code.
Edit: seems to be something for the menu module. Still not sure, why we need it.

File: core/modules/user/user.module [hunk 7]
```diff
@@ -1630,6 +1631,19 @@ function user_menu_alter(&$items) {
...
+
+  if (!isset($user_login_page_tabs)) {
+    $user_login_page_tabs = config_get('system.core', 'user_login_page_tabs');
+  }
+
+  return $user_login_page_tabs;
```
Comment (indigoxela) [inline core/modules/user/user.module:1644]: Seems like the only purpose of this function is to return a single config item.
What was the intention behind making this a full function? What's the benefit?
Config *is* already cached, AFAIK.
At first sight, this appears like over-abstraction to me, but I might miss something. :thinking:

File: core/profiles/standard/standard.install (hunk 6 above)
Comment (docwilmot) [reply core/profiles/standard/standard.install:589]: Menu items that are type `MENU_CALLBACK` don't appear in the menu unless they're customized. I confess I don't quite know why. But without this here, the "Login" link doesnt appear in the Account menu.

File: core/modules/user/user.module (hunk 7 above)
Comment (docwilmot) [reply core/modules/user/user.module:1644]: Thanks for pointing this out. So generally no, config isn't cached, except if the `_config_static` is true in the config file, which it is in the `system.core.json` file. So you're right we actually don't need this function in this case.

File: core/modules/user/user.module (hunk 5 above)
Comment (docwilmot) [reply core/modules/user/user.module:1638]: Agreed not needed, see below comment.

File: core/modules/user/user.module (hunk 4 above)
Comment (docwilmot) [reply core/modules/user/user.module:1420]: I don't understand why the closing parenthesis annoys you, but we probably can remove both opening and closing..

File: core/modules/user/user.admin.inc (hunk 3 above)
Comment (docwilmot) [reply core/modules/user/user.admin.inc:411]: Could do

File: core/modules/user/user.admin.inc (hunk 1 above)
Comment (docwilmot) [reply core/modules/user/user.admin.inc:409]: Could do

File: core/modules/user/user.module (hunk 4 above)
Comment (indigoxela) [reply core/modules/user/user.module:1420]: I really should have been clearer in my comment. :wink: 

What expected is:
```
($user_login_page_tabs) ? MENU_DEFAULT_LOCAL_TASK : MENU_CALLBACK,
```
The result seems OK with both parenthesis positions, but I found it irritating, when wrapped as a whole. (Note: not annoyed, just irritated :wink: ) Maybe that's just me... :shrug:

Comment (herbdool) [reply core/modules/user/user.module:1420]: I don't know if we have standard but I don't see examples in core where the whole thing is in parentheses. And parentheses are only used on the first part if it's doing some comparison.

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/4750.

**Pull request closed without merging**